server/sockets/list.js

```diff
@@ -413,42 +433,62 @@ const changeListType = (socket, dashboardClients, cohortClients, listClients) =>
 
           if (dashboardClients.size > 0) {
             if (type === ListType.LIMITED && removedViewers) {
-              emitForMany(
-                removedViewers,
-                dashboardClients,
-                socket,
-                ListActionTypes.DELETE_SUCCESS,
-                listId
-              );
+              removedViewers.forEach(id => {
+                const userId = id.toString();
+
+                if (dashboardClients.has(userId)) {
+                  socket.broadcast
+                    .to(dashboardClients.get(userId))
+                    .emit(ListActionTypes.DELETE_SUCCESS, listId);
+                }
+              });
             }
 
-            emitListForMany(
-              viewersIds,
-              dashboardClients,
-              socket,
-              ListActionTypes.FETCH_META_DATA_SUCCESS,
-              list
-            );
+            viewersIds.forEach(id => {
+              const userId = id.toString();
+
+              if (dashboardClients.has(userId)) {
```

Comment:
and here wil be no need to check this condition again cause it will be already checked in the first IF

Reply:
I removed checking the size and refactored the code but a little bit different.